core/src/mast/serialization/info.rs

```diff
@@ -51,7 +51,12 @@ impl MastNodeInfo {
                 let left_child = MastNodeId::from_u32_safe(left_child_id, mast_forest)?;
                 let right_child = MastNodeId::from_u32_safe(right_child_id, mast_forest)?;
 
-                Ok(MastNode::new_join(left_child, right_child, mast_forest))
+                MastNode::new_join(left_child, right_child, mast_forest).map_err(|e| {
+                    DeserializationError::InvalidValue(format!(
+                        "Failed to deserialize Join node: {}",
+                        e
+                    ))
+                })
```

Comment:
Here, and in other similar places in this file, we could probably just use `expect()` because validity of the child nodes should be guaranteed by `MastNodeId::from_u32_safe()` constructor in the calls above.